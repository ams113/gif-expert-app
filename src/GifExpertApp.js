import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
    // const categories = ['HTML', 'CSS', 'JAVASCRIPT'];

    const [categories, setCategories] = useState(defaultCategories);

    /* const handleAdd = () => {
        setCategories( categories => [...categories, 'REACT'] );
    }; */


    return (
        <>
            <header>
                <div className="search">
                    <h2>GifExpertApp</h2>
                    <AddCategory setCategories={setCategories} />
                </div>
            </header>
        
            <main div className="container"> 
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </main>
        </>
    )
}
