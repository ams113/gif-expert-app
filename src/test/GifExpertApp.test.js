import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Testing App: GifExpertApp', () => {
    
    test('should renders correctly', () => {

        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should renders category list', () => {

        const categories = ['naruto', 'goku']
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length );
    });

});
