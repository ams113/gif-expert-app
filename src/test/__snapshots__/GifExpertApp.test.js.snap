// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App: GifExpertApp should renders category list 1`] = `
<Fragment>
  <header>
    <div
      className="search"
    >
      <h2>
        GifExpertApp
      </h2>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </header>
  <main
    className="container"
    div={true}
  >
    <GifGrid
      category="naruto"
      key="naruto"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </main>
</Fragment>
`;

exports[`Testing App: GifExpertApp should renders correctly 1`] = `
<Fragment>
  <header>
    <div
      className="search"
    >
      <h2>
        GifExpertApp
      </h2>
      <AddCategory
        setCategories={[Function]}
      />
    </div>
  </header>
  <main
    className="container"
    div={true}
  />
</Fragment>
`;
